superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/FiltersConfigForm.tsx

```diff
@@ -654,7 +655,20 @@ const FiltersConfigForm = (
   useEffect(() => {
     if (datasetId) {
       cachedSupersetGet({
-        endpoint: `/api/v1/dataset/${datasetId}`,
+        endpoint: `/api/v1/dataset/${datasetId}?q=${rison.encode({
+          columns: [
```

Comment:
I found other attributes that need to be queried when analyzing the following files:

https://github.com/apache/superset/blob/da3791ad3daa209631a588394600d1a8b635e814/superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterControl/index.jsx#L138

```
database.id
datasource_name
schema
is_sqllab_view
```

https://github.com/apache/superset/blob/da3791ad3daa209631a588394600d1a8b635e814/superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSimpleTabContent/index.tsx#L121

```
columns.type
```

https://github.com/apache/superset/blob/da3791ad3daa209631a588394600d1a8b635e814/superset-frontend/packages/superset-ui-chart-controls/src/utils/getTemporalColumns.ts#L43

```
columns.is_dttm
columns.name
main_dttm_col
```

https://github.com/apache/superset/blob/0c454c64426376d7fb209a8b16d15c580be811f4/superset-frontend/packages/superset-ui-chart-controls/src/types.ts#L482

```
results
sql
```

Reply:
@michael-s-molina it seem like `results` isn't an attribute returned per [here](https://github.com/apache/superset/blob/master/superset/datasets/api.py#L131-L191).

Reply:
@john-bodley You're right. Given that `results` is not listed in `show_select_columns`, I'm assuming that it's populated elsewhere and you projections won't affect this behavior. Tagging @zhaoyongjie and @villebro in case they know something different.

Reply:
Agreed, I think this is needed in the Explore control panel only, hence shouldn't affect dashboards/native filters.